Recrutify/Recrutify.Services/Services/CandidateService.cs

```diff
@@ -50,12 +50,27 @@ public IQueryable<CandidateDTO> Get()
             return _mapper.ProjectTo<CandidateDTO>(_candidateRepository.Get());
         }
 
-        public async Task<CandidateDTO> CreateAsync(CandidateCreateDTO candidateCreateDTO)
+        public async Task<CandidateDTO> CreateAsync(CandidateCreateDTO candidateCreateDTO, Guid projectId)
         {
             var candidate = _mapper.Map<Candidate>(candidateCreateDTO);
-            await _candidateRepository.CreateAsync(candidate);
-            var result = _mapper.Map<CandidateDTO>(candidate);
-            return result;
+            var currentCandidate = await _candidateRepository.GetByEmailAsync(candidate.Email);
+            if (currentCandidate == null)
+            {
+                candidate.Id = Guid.NewGuid();
+                await _candidateRepository.CreateAsync(candidate);
+                var newCanditate = _mapper.Map<CandidateDTO>(candidate);
+                return newCanditate;
+            }
+
+            var candidateToUpdate = _mapper.Map(candidateCreateDTO, currentCandidate.DeepCopy());
+            var primarySkill = _mapper.Map<CandidatePrimarySkill>(candidateCreateDTO.PrimarySkill);
+            var projectResults = currentCandidate?.ProjectResults?.ToList();
+            var newProjectResult = new ProjectResult { ProjectId = projectId, PrimarySkill = primarySkill };
+            projectResults?.Add(newProjectResult);
```

Comment:
if projectResults are null, check it and init it with new list, with one element newProjectResults, if not null just add new element to it, like you wrote now

Reply:
done